src/main/java/com/btrapp/jklarfreader/objects/KlarfReader12.java

```diff
@@ -503,22 +510,40 @@ protected void readList(KlarfTokenizer kt, KlarfMappingRecord kmr)
         "Klarf list type not supported " + kmr.klarfKey12(), kt, ExceptionCode.ListFormat);
   }
 
-  private List<List<String>> readEmbeddedList(KlarfTokenizer kt)
+  private List<List<String>> readEmbeddedList(KlarfTokenizer kt, String colName)
       throws IOException, KlarfException {
-    // have not seen a case for this yet - see KlarfReader18 readEmbeddedList if needed
+    // System.out.println("col name is " + colName);
+    // This is for the IMAGELIST
     int imageQty = kt.intVal();
     // System.out.println("read embedded list qty is " + imageQty);
     if (imageQty == 0) {
       return Collections.emptyList();
     }
+    List<List<String>> outerList = new ArrayList<>(imageQty);
+    List<String> innerList = new ArrayList<>();
```

Comment:
is it worth specifying the size so it doesn't have to grow like in L522

Reply:
Good idea.  Done.